### DISCUSSION TITLE
How can I Load Theme and Variables from an API?

### DISCUSSION THREAD

**Discussion by ankursehdev** [tailwindlabs/tailwindcss#12333] [Help]
Hello All, 

We are building a _Design System where Tokens are generated through Figma and published_, which we can download via rest API. 

As per the docs, we can customize the theme according to our requirements but 

**Is it possible to download `variables` and `theme config` from a rest endpoint and then merge it in `:root` somehow?**

![tailwind](https://github.com/tailwindlabs/tailwindcss/assets/34501847/0ec0c2ad-4fce-4cea-923c-a3ad66a1c391)

**Comment by wongjn** [comment]
You could consider using CSS variables as values in `tailwind.config.js`, and then have these CSS variables be injected into the page at runtime.

**Comment by Erij-Maherzia-BEN-BRAHIM** [comment]
hi @ankursehdev 
in your `tailwind.config.js` set the property **darkMode** by the value **class**, then in the root of your app call the API, for example, your API returns `dataTheme`, and this response could be `dark` or `light`, you can use it this way 👇🏻 
```html
<html
      className={dataTheme}
      data-theme={dataTheme}
    >
  // rest of your code
 </html>
```

**Comment by gagadzq** [comment]
Have you solved it yet? I'm facing the same problem
